server/issue.go

```diff
@@ -83,6 +83,23 @@ func httpAPICreateIssue(ji Instance, w http.ResponseWriter, r *http.Request) (in
 		Fields: &create.Fields,
 	}
 
+	for i, notCovered := range create.RequiredFieldsNotCovered {
+
+		if notCovered == "reporter" || notCovered == "Reporter" {
+			requiredFieldsNotCovered := create.RequiredFieldsNotCovered[:i]
+			if i < len(create.RequiredFieldsNotCovered) {
+				requiredFieldsNotCovered = append(requiredFieldsNotCovered,
+					create.RequiredFieldsNotCovered[i+1:]...)
+			}
+			create.RequiredFieldsNotCovered = requiredFieldsNotCovered
+
+			if ji.GetType() == JIRATypeServer {
+				issue.Fields.Reporter = &jiraUser.User
```

Comment:
So we don't just want this line because:
- We want to support changing the reporter in some circumstances.
- Jira cloud barfs if we set it?

Correct?

Reply:
Yes. Jira Cloud barfs at it. And on the Server we end up only setting it if it is listed as "required", otherwise it's empty (and defaults to the user). I am hesitant to just force it atter seeing it fail on the Cloud.

Reply:
@levb what happens on Jira Cloud then?

Reply:
@jasonblais I'll test some more, but so far everything appeared to work. mmtest has some weird field setup, so I couldn't really test there, worked for me on `levbrouk`. I suggest that any cloud-specific issues that arise we'd have to address later as bugs. This is still an improvement for the server.

Reply:
Thanks @levb, that sounds good then. We can also have Dylan smoke test post-merge.